src/Compilers/CSharp/Portable/Emitter/Model/TypeParameterSymbolAdapter.cs

```diff
@@ -292,7 +293,8 @@ bool Cci.IGenericParameter.MustHaveDefaultConstructor
             {
                 //  add constructor constraint for value type constrained 
                 //  type parameters to comply with Dev11 output
-                return this.HasConstructorConstraint || this.HasValueTypeConstraint;
+                //  do this for "unmanaged" constraint too
+                return this.HasConstructorConstraint || this.HasValueTypeConstraint || this.HasUnmanagedTypeConstraint;
```

Comment:
Is this checked in test?

Reply:
indirectly, yes. Added a direct testcase as well.

---
In reply to: [173662727](https://github.com/dotnet/roslyn/pull/25397#discussion_r173662727) [](ancestors = 173662727)